backend/src/main/route/PromotionRouter.ts

```diff
@@ -0,0 +1,21 @@
+import express, { Router } from 'express';
+import { PromotionController } from '../controller/PromotionController';
+
+export class PromotionRouter {
+  private promotionRouter = express.Router();
+  private promotionController = new PromotionController();
+  constructor(promotionController: PromotionController) {
+    this.promotionController = promotionController;
+  }
+
+  getRoutes(): Router {
```

Comment:
@yunishimura0716 let me know if there's anything in this PR you'd want to follow a different design, for example the routers, controllers, services, validation etc. I know you are working on the `User`'s endpoints at the moment and it will take some coordination from both of us to make sure our PRs follow the same format and don't have merge conflicts. 

@andrewzulaybar 
I did not implement the following endpoints from the meeting notes. I feel like these are endpoints meant for the `UserController`. We also need an endpoint to `GET` the saved promotions.
```
POST /promotion/:promotion_id/save
DELETE /promotion/:promotion_id/save
```

I was thinking more along the lines of:
```
POST /users/:id/savedPromotions
DELETE /users/:id/savedPromotions
GET /users/:id/savedPromotions
```

Reply:
@ericrkuo That sounds good! So like the request body to `POST` and `DELETE` `/users/:id/savedPromotions` would include the promotion_id? Also another thing that I'm not sure I considered earlier, have we designed a way to get the number of saves for a specific promotion?

Reply:
yup! so promotionId would be a request query, and id would be request param.

Yu is working on it currently, but I remember when I was playing around with it, you could do this:
![image](https://user-images.githubusercontent.com/49849754/99343308-7624ae00-2842-11eb-8cb4-6d965ae6e224.png)

@yunishimura0716 the very tricky part is getting the `savedAt` timestamp from the `saved_promotion` table. I think I have an idea of how to do this.
1. use saved promotion repository to get all entries where userId is the desired id. (do not do any joins, we just need the promotionId's and the savedAt timestamp)
2. use all the promotionIds from 1, and execute query to get all promotions + join with discount where id is in the id's from 1.
3. then you can probably do something like
```
return promotions.map((promotion) => return {
  ...promotion,
  savedAt: <the respective timestamp>
}
```
 The reason why you cannot just execute query is because the data won't be formatted the way we want to
we want:
```
we want:
{
  id: 
  name: 
  discount: {
     type:
  }
  savedAt:
}

but we'll actaully get 

{
  id: 
  name: 
  discount_type:
  discount_discountValue:
  savedAt
}
```

This is because if you want to execute any sql query that doesn't return a promotion (because the promotion entity doesn't have a column `savedAt`), you need to call `getRawMany`, and it won't be formatted the way we want it to. This might not make sense, but I think you'll encounter this problem when you try to get the `savedAt` time stamp.

Here's how I did it for full text search, but I added a private property to `Promotion` called `rank` instead of doing the `.map(promotion => return {...promotion, savedAT: ...})` approach.